sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/DataSourceV2Utils.scala

```diff
@@ -117,7 +119,24 @@ private[sql] object DataSourceV2Utils extends Logging {
           hasCatalog,
           catalogManager,
           dsOptions)
-        (catalog.loadTable(ident), Some(catalog), Some(ident))
+
+        val version = hasCatalog.extractTimeTravelVersion(dsOptions)
+        val timestamp = hasCatalog.extractTimeTravelTimestamp(dsOptions)
```

Comment:
Can't we convert them into scala option and then simply call `TimeTravelSpec.create`?

Reply:
Fixed